src/mondo_robot_template.py

```diff
@@ -49,20 +49,32 @@ def run(input_file: str = INPUT_FILE, output_file: str = OUTPUT_FILE):
     df = pd.read_csv(input_file, sep='|').rename(columns={'#CUI': 'xref_id'})
 
     # Get explicit Medgen (CUI, CN) -> Mondo mappings
-    df_medgen_mondo = df[df['source'] == 'MONDO'][['source_id', 'xref_id']].rename(columns={'source_id': 'mondo_id'})
-    out_df_cui_cn = pd.DataFrame(_prefixed_id_rows_from_common_df(df_medgen_mondo))
+    df_medgen_mondo = df[df['source'] == 'MONDO'][['source_id', 'xref_id']].rename(
+        columns={'source_id': 'mondo_id', 'xref_id': 'umls_cui'})
+    df_medgen_mondo['umls_cui'] = df_medgen_mondo['umls_cui'].apply(lambda x: f'UMLS:{x}')
+    # todo: remove
+    # out_df_cui_cn = pd.DataFrame(_prefixed_id_rows_from_common_df(df_medgen_mondo))
 
     # Get Medgen (UID) -> Mondo mappings
     # - Done by proxy: UID <-> CUI <-> MONDO
-    df_medgen_medgenuid = df[df['source'] == 'MedGen'][['source_id', 'xref_id']].rename(
-        columns={'source_id': 'medgen_uid'})
-    out_df_uid = pd.merge(df_medgen_mondo, df_medgen_medgenuid, on='xref_id').rename(
-        columns={'xref_id': 'source_id', 'medgen_uid': 'xref_id'})[['mondo_id', 'xref_id', 'source_id']]
-    out_df_uid['xref_id'] = out_df_uid['xref_id'].apply(lambda x: f'MEDGEN:{x}')
-    out_df_uid['source_id'] = out_df_uid['source_id'].apply(lambda x: f'UMLS:{x}')
+    df_umls_medgenuid = df[df['source'] == 'MedGen'][['source_id', 'xref_id']].rename(
+        columns={'source_id': 'medgen_uid', 'xref_id': 'umls_cui'})
+    df_umls_medgenuid['umls_cui'] = df_umls_medgenuid['umls_cui'].apply(lambda x: f'UMLS:{x}')
+    df_umls_medgenuid['medgen_uid'] = df_umls_medgenuid['medgen_uid'].apply(lambda x: f'MEDGEN:{x}')
+    df_merged = pd.merge(df_medgen_mondo, df_umls_medgenuid, on='umls_cui')
+    # - Split into (Mondo <-> Medgen UID) & (Mondo <-> UMLS CUI)
+    out_df_medgenuid = df_merged.rename(columns={'medgen_uid': 'xref_id', 'umls_cui': 'source_id'})[[
+        'mondo_id', 'xref_id', 'source_id']]
+    out_df_medgenuid['source_id'] = ''
+    out_df_umlscui = df_merged.rename(columns={'umls_cui': 'xref_id', 'medgen_uid': 'source_id'})
+    # - Combine
+    out_df = pd.concat([out_df_medgenuid, out_df_umlscui]).sort_values(['xref_id', 'mondo_id']).drop_duplicates().fillna('')
```

Comment:
## Flipped xref->source namespaces to: UMLS->MedGen
Previously I had set up `xref_id` such that there were MEDGEN terms with UMLS as `source_id`, and the rows where the `xref_id` was UMLS had no `source_id`. Now, per #15 it is flipped.

I trust that @matentzn has the best, most up-to-date information about this. And this decision is also consistent with the mapping set, in light of the fact that there are rows in `MedGenIDMappings.txt` where the `source`=MedGen, but none where the source is UMLS.

Maybe I misunderstood what I learned from Meagan, etc, but the reason that I had done it this way is because I had heard that basically UMLS came first / UMLS IDs were primary, and that MedGen came after UMLS and ingested those IDs. If there was no UMLS ID, it created a CN (CUI Novel). And then, further, I had heard mixed things about UIDs (the ones without a C or CN at the beginning of them, which we apply the prefix "MEDGEN" to). But one thing I heard was that they were extra IDs, primarily used for internal use, and unstable. Anyhow, based on what we're doing now, I assume that must be incorrect.

Reply:
I will tell you based on the result - my understanding is that the MedGen IDs are stable in the sense that no same ID will ever refer to more than one disease, but unstable in the sense that if a disease has one medgen id today it may have another tomorrow. That is ok.

Reply:
That's good to know! I'll add that to my notes.